### PR TITLE
feat(proxy): use squid as NAT instance

### LINKED ISSUE #55: replace internet gateway with nginx proxy for k8s

### PR DESCRIPTION
**Pull request by philloooo**
resolve #55 
- removed NAT gateway
- moved tf_variables and tfstate to `~/.creds/{vpc_name}` so that I can use cloud-automation with different vpcs locally
- used envsubst instead of sed for tf_variables templating
- some minor fixes

tested in OCC with `pr65` vpc name :p

r? @zflamig

### CODE REVIEW

File: pack_and_terraform.bash [hunk 1]
```diff
@@ -1,10 +1,17 @@
 #!/bin/bash
 # Right now this script guides you through the process
 # of running Packer & Terraform
-
+unamestr=`uname`
+if [[ "$unamestr" == 'Darwin' ]]; then
+    export PATH=${PATH}:/usr/local/opt/gettext/bin
+    if [[ ! -x "$(command -v envsubst)" ]]; then
+        brew install gettext
```
Comment (zflamig) [inline pack_and_terraform.bash]: Can we not do brew (I don't have it)? Lets either compile from source or just error out if missing and give the user a couple of options.

File: pack_and_terraform.bash [hunk 2]
```diff
@@ -79,78 +86,79 @@ if echo "$RUNTF" | grep -iq "^y"; then
...
+    export SSHKEY=$SSHKEY
+    export ADDSSHKEY=$SSHKEY
+    export BUCKET=$BUCKET
+    export CLIENTSECRET=$CLIENTSECRET
+    export CLIENTID=$CLIENTID
+    export HMAC=$HMAC
+    export USERAPIDBPASS=$USERAPIDBPASS
+    export INDEXDDBPASS=$INDEXDDBPASS
+    export GDCAPIDBPASS=$GDCAPIDBPASS
+    export INDEXD=$INDEXD
```
Comment (zflamig) [inline pack_and_terraform.bash]: Can we pass these just to envsubst so we don't pollute the env variables globally?

File: setup_kube.bash [hunk 3]
```diff
@@ -1,13 +1,18 @@
...
 cd tf_files
-LOGIN_NODE=`grep -A20 "aws_eip.login" terraform.tfstate | grep "public_ip" | head -1 | sed 's/[ \",]//g' | cut -d: -f2`
+LOGIN_NODE=`grep -A20 "aws_eip.login" $creds_dir/terraform.tfstate | grep "public_ip" | head -1 | sed 's/[ \",]//g' | cut -d: -f2`
 
 echo "Working with Login Node: $LOGIN_NODE"
 
-OUTPUT_DIR=`ls -d *_output | head -1`
 echo "Working with $OUTPUT_DIR"
-VPCNAME=`echo $OUTPUT_DIR | cut -d_ -f1`
+OUTPUT_DIR=${VPC_NAME}_output
```
Comment (zflamig) [inline setup_kube.bash]: This should be above the echo

File: pack_and_terraform.bash (hunk 2 above)
Comment (philloooo) [inline pack_and_terraform.bash]: sure, why do you care as we only have envsubst and terraform in the last though? it won't pollute anything outside of the script

**Pull request merged**